include/translatable-objects.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Registry for all translatable objects.
+ *
+ * @since 3.4
+ *
+ * @phpstan-implements IteratorAggregate<non-empty-string, PLL_Translatable_Object>
+ */
+class PLL_Translatable_Objects implements IteratorAggregate {
+
+	/**
+	 * List of registered objects.
+	 *
+	 * @var PLL_Translatable_Object[] Array keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-var array<non-empty-string, PLL_Translatable_Object>
+	 */
+	private $objects = array();
+
+	/**
+	 * Registers a translatable object.
+	 *
+	 * @since 3.4
+	 *
+	 * @param PLL_Translatable_Object $object The translatable object to register.
+	 * @return PLL_Translatable_Object
+	 */
+	public function register( PLL_Translatable_Object $object ) {
+		if ( ! isset( $this->objects[ $object->get_type() ] ) ) {
+			$this->objects[ $object->get_type() ] = $object;
+		}
+
+		return $this->objects[ $object->get_type() ];
+	}
+
+	/**
+	 * Returns all registered translatable objects.
+	 *
+	 * @since 3.4
+	 *
+	 * @return ArrayIterator Iterator on $objects array property. Keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-return ArrayIterator<string, PLL_Translatable_Object>
```

Comment:
Replacing `non-empty-string` by `string` here will probably cause other PHPStan issues elsewhere.
Let's keep it like that for now, we'll see what happens in the future big PHPStan sweep.